### DISCUSSION TITLE
Device Kernel Image is Invalid (v1.2.1)

### DISCUSSION THREAD

**Discussion by AustinDoolittle** [apache/incubator-mxnet#12228] [Q&A]
## Description
Unable to allocate any GPU memory when using mxnet 1.2.1 with Cuda Versions 9.0-.2

## Environment info (Required)
OS: Windows 10 Enterprise
CPU: Intel Core i7-6800K 
GPU: Nvidia GTX 1060 and Nvidia GTX 1070
Mxnet Version: 1.2.1, installed via pip install mxnet-cu90/mxnet-cu91/mxnet-cu92
Cuda Version: 9.0-.2

```
----------Python Info----------
Version      : 3.6.6
Compiler     : MSC v.1900 64 bit (AMD64)
Build        : ('default', 'Jun 28 2018 11:27:44')
Arch         : ('64bit', 'WindowsPE')
------------Pip Info-----------
Version      : 10.0.1
Directory    : C:\tools\Anaconda3\envs\mxnet_dev_env\lib\site-packages\pip
----------MXNet Info-----------
Version      : 1.2.1
Directory    : C:\tools\Anaconda3\envs\mxnet_dev_env\lib\site-packages\mxnet
Hashtag not found. Not installed from pre-built package.
----------System Info----------
Platform     : Windows-10-10.0.15063-SP0
system       : Windows
node         : [redacted]
release      : 10
version      : 10.0.15063
----------Hardware Info----------
machine      : AMD64
processor    : Intel64 Family 6 Model 79 Stepping 1, GenuineIntel
Name
Intel(R) Core(TM) i7-6800K CPU @ 3.40GHz
```

Package used (Python/R/Scala/Julia): Python

## Error Message:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\tools\Anaconda3\envs\mxnet_dev_env\lib\site-packages\mxnet\ndarray\utils.py", line 146, in array
    return _array(source_array, ctx=ctx, dtype=dtype)
  File "C:\tools\Anaconda3\envs\mxnet_dev_env\lib\site-packages\mxnet\ndarray\ndarray.py", line 2338, in array
    arr = empty(source_array.shape, ctx, dtype)
  File "C:\tools\Anaconda3\envs\mxnet_dev_env\lib\site-packages\mxnet\ndarray\ndarray.py", line 3548, in empty
    return NDArray(handle=_new_alloc_handle(shape, ctx, False, dtype))
  File "C:\tools\Anaconda3\envs\mxnet_dev_env\lib\site-packages\mxnet\ndarray\ndarray.py", line 139, in _new_alloc_handle
    ctypes.byref(hdl)))
  File "C:\tools\Anaconda3\envs\mxnet_dev_env\lib\site-packages\mxnet\base.py", line 149, in check_call
    raise MXNetError(py_str(_LIB.MXGetLastError()))
mxnet.base.MXNetError: [16:54:01] c:\jenkins\workspace\mxnet-tag\mxnet\src\storage\pooled_storage_manager.h:108: cudaMalloc failed: device kernel image is invalid

## Minimum reproducible example
```
import mxnet as mx
arr = mx.nd.array([0], ctx=mx.gpu())
```
OR
```
import mxnet as mx
arr = mx.nd.array([0])
arr.as_in_context(mx.gpu())
```

## What have you tried to solve it?

1. Installed Cuda 9.0, 9.1, and 9.2 (with corresponding mxnet binaries)
2. Installed a second graphics card (allocating to both the 1060 and 1070 do not work)
3. Tried allocating pytorch tensors to the GPU and was successful
4. Ultimately downgraded to mxnet 1.2.0 and this resolved the issue.

**Comment by vdantu** [comment]
@AustinDoolittle : Thanks for raising this issue. We will look into it. Just to make things clearer, did you uninstall and reinstall the CUDA drivers and then the SDK?  

@mxnet-label-bot : [Python, Build, CUDA, Windows, Question]

**Comment by AustinDoolittle** [comment]
@vdantu That is correct, I would uninstall the current version of cuda, install either the came version or a different version (I tried 9.0, 9.1, and 9.2), then install the matching mxnet version via pip. One thing that may be important to mention is that this started happening after I had encountered a BSOD. I don't think I was running any mxnet processes when this occurred, but I could be mistaken.

**Comment by AustinDoolittle** [comment]
Ok, I was able to resolve this issue. The steps to correct were:

1. Uninstall everything remotely related to nvidia (drivers, cuda, cudnn, documentation, etc.)
2. Uninstall MxNet
3. Reboot
4. Install nvidia drivers
5. Reboot
6. Install Cuda and Cudnn
7. Reboot
8. Install MxNet

I think I definitely went a little overkill with the rebooting, but everything appears to be working now. Thanks for the assistance!

**Comment by zhenyu** [comment]
I encounter this issue ,and try with @AustinDoolittle method, not works.
either of the following two works:
1)pip install mxnet-cu92==1.2.0
2)pip install mxnet-cu92mkl
So I think it should be bug

**Comment by AustinDoolittle** [comment]
Before I was able to fully resolve the bug with 1.2.1, I did find success like @zhenyu by reverting to mxnet-cu92=1.2.0

**Comment by AustinDoolittle** [comment]
This issue cropped up again for some reason. I took @zhenyu's advice and installed mxnet-cu92mkl. So far it appears to be working.

**Comment by hitdongfeng** [comment]
I have had the same problem as you and tried many times without success. I think it might be a bug

**Comment by tianxingyzxq** [comment]
I have the same problem with mxnet-cu92

**Comment by hitdongfeng** [comment]
Uninstalling version 1.2.1 and installing version 1.2.0 will fix the problem.

**Comment by tianxingyzxq** [comment]
@hitdongfeng mxnet 1.2.0 is ok now. thanks.

**Comment by zhenyu** [comment]
Just another reminder, I have installed MKL previously, I believe it is related to the mlk instead of normal version of mxnet cu92

**Comment by zhangyichent** [comment]
mxnet-cu92  1.3.0 is error
mxnet-cu92  1.2.0 is ok

**Comment by AustinDoolittle** [comment]
Any updates on this?
